apps/andi/lib/andi_web/live/edit_live_view.ex

```diff
@@ -34,383 +39,148 @@ defmodule AndiWeb.EditLiveView do
        save_success: false,
        success_message: "",
        test_results: nil,
        testing: false,
-       dataset_exists: dataset_exists,
+       finalize_form_data: nil,
        unsaved_changes: false,
        show_unsaved_changes_modal: false
-     )
-     |> assign(get_default_dictionary_field(new_changeset))}
-  end
-
-  def handle_event("test_url", _, socket) do
-    changes = Ecto.Changeset.apply_changes(socket.assigns.changeset)
-    technical = Map.get(changes, :technical)
-    url = Map.get(technical, :sourceUrl) |> Andi.URI.clear_query_params()
-    query_params = key_values_to_keyword_list(technical, :sourceQueryParams)
-    headers = key_values_to_keyword_list(technical, :sourceHeaders)
-
-    Task.async(fn ->
-      {:test_results, Andi.Services.UrlTest.test(url, query_params: query_params, headers: headers)}
-    end)
-
-    {:noreply, assign(socket, testing: true)}
-  end
-
-  def handle_event("validate", %{"form_data" => form_data, "_target" => ["form_data", "technical", "sourceUrl"]}, socket) do
-    form_data
-    |> FormTools.adjust_source_query_params_for_url()
-    |> InputConverter.form_data_to_ui_changeset()
-    |> complete_validation(socket)
-    |> mark_changes()
-  end
-
-  def handle_event("validate", %{"form_data" => form_data, "_target" => ["form_data", "technical", "sourceQueryParams" | _]}, socket) do
-    form_data
-    |> FormTools.adjust_source_url_for_query_params()
-    |> InputConverter.form_data_to_ui_changeset()
-    |> complete_validation(socket)
-    |> mark_changes()
-  end
-
-  def handle_event(
-        "validate",
-        %{"form_data" => form_data, "_target" => ["form_data", "business", "dataTitle" | _]},
-        %{assigns: %{dataset_exists: false}} = socket
-      ) do
-    form_data
-    |> FormTools.adjust_data_name()
-    |> InputConverter.form_data_to_ui_changeset()
-    |> complete_validation(socket)
-    |> mark_changes()
-  end
-
-  def handle_event(
-        "validate",
-        %{"form_data" => form_data, "_target" => ["form_data", "technical", "orgId" | _]},
-        %{assigns: %{dataset_exists: false}} = socket
-      ) do
-    form_data
-    |> FormTools.adjust_org_name()
-    |> InputConverter.form_data_to_ui_changeset()
-    |> Dataset.validate_unique_system_name()
-    |> complete_validation(socket)
-    |> mark_changes()
-  end
-
-  def handle_event("validate", %{"form_data" => form_data}, socket) do
-    form_data
-    |> InputConverter.form_data_to_ui_changeset()
-    |> complete_validation(socket)
-    |> mark_changes()
+     )}
   end
 
-  def handle_event("validate_system_name", _, socket) do
-    changeset =
-      Dataset.validate_unique_system_name(socket.assigns.changeset)
-      |> Map.put(:action, :update)
+  def handle_info(:publish, socket) do
+    socket = reset_save_success(socket)
 
-    {:noreply, assign(socket, changeset: changeset)}
-  end
+    AndiWeb.Endpoint.broadcast_from(self(), "form-save", "save-all", %{})
+    Process.sleep(1_000)
```

Comment:
Right now, this broadcasts a message to all components to save their form before we attempt to publish. I'm open to other ideas if anyone has an objection to the asynchronous behavior or the sleep

Reply:
If you've researched it pretty thoroughly I'm up for letting this be the pattern for now.